src/mapping.ts

```diff
@@ -10,18 +10,37 @@ import { tokens, trades, orders, users } from "./modules"
 export function handleInteraction(event: Interaction): void {}
 
 export function handleOrderInvalidated(event: OrderInvalidated): void {
+
   let orderId = event.params.orderUid.toHexString()
-  orders.invalidateOrder(orderId)
+  let timestamp = event.block.timestamp
+
+  let order = orders.invalidateOrder(orderId, timestamp)
+
+  order.save()
 }
 
-export function handlePreSignature(event: PreSignature): void {}
+export function handlePreSignature(event: PreSignature): void {
+
+  let orderUid = event.params.orderUid.toHexString()
+  let ownerAddress = event.params.owner
+  let owner = ownerAddress.toHexString()
+  let signed = event.params.signed
```

Comment:
Isn't this implied in this event? Can it ever be `false`?

Reply:
I think it's implied in this event but there's a param called signed. That's why I thought it would be a good idea to add it there as param value instead of a constant. I'm not sure if there's a scenario where it could be false, :thinking:

Reply:
Do you want me to set ```true``` value?

Reply:
I rather not have the `signed` flag, since a preSign even should always mean it's signed

Reply:
removed! redeployed in gabrielcamba/cow-staging

Reply:
I think we either need the param, or we need to use it for setting or removeing the timestamp. 
Please see 
https://github.com/gnosis/cow-subgraph/pull/1#discussion_r785349312